Mathlib/Algebra/BigOperators/Group/Finset/Basic.lean

```diff
@@ -618,6 +618,19 @@ theorem prod_range_induction (f s : ℕ → M) (base : s 0 = 1)
   | zero => rw [Finset.prod_range_zero, base]
   | succ k hk => simp only [hk, Finset.prod_range_succ, step, mul_comm]
 
+/-- A version of `Finset.prod_range_induction` where the ratios of adjacent terms only needs to be
+checked for terms up to `n` -/
+@[to_additive "A version of `Finset.sum_range_induction` where the differences of adjacent terms
+only needs to be checked for terms up to `n`"]
+theorem prod_range_induction' (f s : ℕ → M) (base : s 0 = 1)
```

Comment:
Have you tried to replace the original lemma with this one? Does need lots of tiny fixes in the library? If not, let's just only have this version (but without the `'` in the name).